### PR TITLE
Change to new graphic test strategy - jobs and test plans (BugFix)

### PR DESCRIPTION
**Pull request by hanhsuan**
## Description
This is part of #586 that includes the changes of job and test plan only.

1. New jobs that uses prime_offload_tester.py to valid GPU rendering.
2. New test plans that conbimes integrated and discrete GPU into one.
3. Remove unnecessary jobs and test plans after switching to new graphic test strategy.

This PR depends on #586.
if #586 isn't merged yet, please don't merge this one.

### CODE REVIEW

File: providers/base/units/graphics/jobs.pxu
Comment (pieqq) [inline providers/base/units/graphics/jobs.pxu:320]:
```suggestion
summary:
    Test that glxgears works for {vendor} {product} (manual)
description:
    Run glxgears while offloading rendering to the {vendor} {product} GPU
    and ask tester to validate the outcome.
```

File: providers/base/units/graphics/jobs.pxu
Comment (pieqq) [inline providers/base/units/graphics/jobs.pxu:343]:
```suggestion
summary:
    Test that glxgears works in full screen mode for {vendor} {product} (automated)
description:
    Run glxgears in full screen mode while offloading rendering to the {vendor}
    {product} GPU and validate the outcome automatically.
```

File: providers/base/units/graphics/jobs.pxu
Comment (pieqq) [inline providers/base/units/graphics/jobs.pxu:358]:
```suggestion
summary:
    Test that glxgears works in full screen mode for {vendor} {product} (manual)
description:
    Run glxgears in full screen mode while offloading rendering to the {vendor}
    {product} GPU and as the tester to validate the outcome.
```

File: providers/base/units/graphics/jobs.pxu
Comment (pieqq) [inline providers/base/units/graphics/jobs.pxu:365]:
```suggestion
    1. Did the 3D animation appear?
```

**Review by pieqq — APPROVED** [review]
Thanks for splitting up the work related to jobs and test plans, it males it easier to review!

I just made a few suggestions (you can accept them directly from within GitHub and it will create git commits automatically).

Otherwise, I think it's ready to land after #586 lands!

**Comment by hanhsuan** [comment]
Thanks for your help !!!

**Comment by hanhsuan** [comment]
@pieqq  After digging into [this issue](https://warthogs.atlassian.net/browse/OEMQA-3877?focusedCommentId=388179), I would like to change this PR to draft and make the jobs could match what gpu-manager do for Ubuntu.

**Comment by hanhsuan** [comment]
We are going to separate the test cases of laptop and desktop and the reason is described  in #1027.

### submissions:
  - laptop:
    - [auto test](https://certification.canonical.com/hardware/202001-27667/submission/360073/)
    - [manual test](https://certification.canonical.com/hardware/202001-27667/submission/360075/)
  - desktop:
    - [auto test](https://certification.canonical.com/hardware/202401-33394/submission/360071/)
    - [manual test](https://certification.canonical.com/hardware/202401-33394/submission/360072/)

**Comment by hanhsuan** [comment]
@pieqq  According to #1009, should I add template id and modify the test plan to use it?

File: providers/base/units/graphics/jobs.pxu
Comment (pieqq) [inline providers/base/units/graphics/jobs.pxu:300]:
```suggestion
    dmi.product in ['Notebook','Laptop','Portable','Convertible']
```

File: providers/base/units/graphics/jobs.pxu
Comment (pieqq) [inline providers/base/units/graphics/jobs.pxu:320]:
```suggestion
    dmi.product in ['Notebook','Laptop','Portable','Convertible']
```

File: providers/base/units/graphics/jobs.pxu
Comment (pieqq) [inline providers/base/units/graphics/jobs.pxu:348]:
```suggestion
    dmi.product in ['Notebook','Laptop','Portable','Convertible']
```

File: providers/base/units/graphics/jobs.pxu
Comment (pieqq) [inline providers/base/units/graphics/jobs.pxu:368]:
```suggestion
    dmi.product in ['Notebook','Laptop','Portable','Convertible']
```

File: providers/base/units/graphics/jobs.pxu
Comment (pieqq) [inline providers/base/units/graphics/jobs.pxu:394]:
```suggestion
    dmi.product in ['Desktop','Low Profile Desktop','Tower','Mini Tower','Space-saving','All In One','All-In-One','AIO']
```

File: providers/base/units/graphics/jobs.pxu
Comment (pieqq) [inline providers/base/units/graphics/jobs.pxu:415]:
```suggestion
    dmi.product in ['Desktop','Low Profile Desktop','Tower','Mini Tower','Space-saving','All In One','All-In-One','AIO']
```

**Review by pieqq — CHANGES_REQUESTED** [review]
My understanding is that you want to test things differently whether it's a laptop or a desktop.

However, for the laptop cases, you use

```
dmi.product in ['Notebook','Laptop','Portable','All In One','All-In-One','AIO','Convertible']
```

I think `'All In One','All-In-One','AIO'` belong to the desktop field. There are the kind of devices that embeds the motherboard into the screen, like an iMac for instance.

So I think you should review the requirements for the jobs you modified in this commit.

**Comment by hanhsuan** [comment]
@pieqq  The reason to add AIO into prime offload test is that the behavior of graphic is the same with laptop. Some AIO will connect embedded monitor with iGPU and provide dGPU for performance mode. Therefore, we would like to cover this kind of condition.

File: providers/base/units/graphics/jobs.pxu
Comment (pieqq) [inline providers/base/units/graphics/jobs.pxu:408]: The `description` field is not required when there is a `purpose` field, so I suggest to merge both:

```suggestion
purpose:
     Tests the basic 3D capabilities of your current video card. This test covers all devices without an integrated display, such as desktops.
```

File: providers/base/units/graphics/jobs.pxu
Comment (pieqq) [inline providers/base/units/graphics/jobs.pxu:432]: Same suggestion as above.

```suggestion
purpose:
     Tests the basic full screen 3D capabilities of your current video card. This test covers all devices without an integrated display, such as desktops.
```

File: providers/base/units/graphics/jobs.pxu
Comment (pieqq) [inline providers/base/units/graphics/jobs.pxu:311]:
```suggestion
    This test covers all devices with an integrated display, such as laptops and all-in-one (AIO) devices.
```

File: providers/base/units/graphics/jobs.pxu
Comment (pieqq) [inline providers/base/units/graphics/jobs.pxu:332]:
```suggestion
    This test covers all devices with an integrated display, such as laptops and all-in-one (AIO) devices.
```

File: providers/base/units/graphics/jobs.pxu
Comment (pieqq) [inline providers/base/units/graphics/jobs.pxu:361]:
```suggestion
    This test covers all devices with an integrated display, such as laptops and all-in-one (AIO) devices.
```

File: providers/base/units/graphics/jobs.pxu
Comment (pieqq) [inline providers/base/units/graphics/jobs.pxu:382]:
```suggestion
    This test covers all devices with an integrated display, such as laptops and all-in-one (AIO) devices.
```

**Review by pieqq — CHANGES_REQUESTED** [review]
Some rewording suggestions. Also, jobs with a `purpose` field don't need a `description` field, so I proposed changes to merge both with a simplified explanation.

**Review by pieqq — APPROVED** [review]
+1

Thanks a lot for your determination!

**Pull request merged**
